unit-tests/src/test/java/org/eclipse/collections/impl/list/primitive/LongIntervalTest.java

```diff
@@ -251,6 +251,66 @@ public void each()
         assertEquals(list2, interval2);
     }
 
+    @Test
+    public void eachForLargeValues()
+    {
+        var values1 = LongLists.mutable.empty();
+
+        LongInterval.zeroToBy(9_999_999_999L, 1_000_000_000L).forEach(values1::add);
+
+        assertEquals(
+                LongLists.immutable.of(
+                        0, 1_000_000_000L, 2_000_000_000L, 3_000_000_000L, 4_000_000_000L,
+                        5_000_000_000L, 6_000_000_000L, 7_000_000_000L, 8_000_000_000L, 9_000_000_000L),
+                values1);
+
+        var values2 = LongLists.mutable.empty();
+
+        LongInterval.zeroToBy(-9_999_999_999L, -1_000_000_000L).forEach(values2::add);
+
+        assertEquals(
+                LongLists.immutable.of(
+                        0, -1_000_000_000L, -2_000_000_000L, -3_000_000_000L, -4_000_000_000L,
+                        -5_000_000_000L, -6_000_000_000L, -7_000_000_000L, -8_000_000_000L, -9_000_000_000L),
+                values2);
+    }
+
+    @Test
+    public void forEachWithIndexForLargeValues()
+    {
+        var values1 = LongLists.mutable.empty();
+        var indices1 = LongLists.mutable.empty();
```

Comment:
It would be clearer if the indices1 were a `MutableIntList`.

There is also another method you changed named `forEachWithLongIndex` which would make perfect sense for this test.

Reply:
`forEachWithLongIndex` is not used and deprecated for removal (that was the change 😀) so I would rather not used it. Will change indices to `MutableIntList`.